### DISCUSSION TITLE
interruptible(boolean) reads differently than its intent

### DISCUSSION THREAD

**Discussion by etspaceman** [typelevel/cats-effect#2248] [Ideas]
I'm wondering if we should create an alias around this. For example,  `interruptible(false)` reads like it isn't interruptible. Perhaps we should consider `interruptibleOnce` and `interruptibleMany`.

**Comment by vasilmkd** [comment]
I think adding `interruptibleOnce` and `interruptibleMany` is indeed the easiest way to introduce this change, without pushing the ergonomics of enums in Scala 2 (`sealed trait InterruptibleMode` and `case object Many`, `case object Once`). I do see this as a positive change and I think we should do it.

**Comment by djspiewak** [comment]
I generally agree this would be a good change, though I wish we could use `interruptible` and `interruptibleMany` instead. Technically we *can* it would just require a pretty major source compatibility breakage.

**Reply by vasilmkd** [reply]
I wish that too but I'm against source breakage and the hassle. Maybe later down the line. I would like for us to go ahead with `interruptibleOnce` and `interruptibleMany` for now.

**Reply by djspiewak** [reply]
I think the counter argument is that the "many" version is quite weird and unlikely to be commonly necessary, thus it would be better to carrot people to the asymmetrical default. It's also not totally clear to me how common the method is in the wild, so we may have a window of opportunity to improve it without hideous pain and suffering.

**Reply by vasilmkd** [reply]
Well put. I do agree with this very much.

**Reply by vasilmkd** [reply]
Do you want me to attempt a scalafix migration too?

**Reply by djspiewak** [reply]
> Do you want me to attempt a scalafix migration too?

Very much yes! It seems nicely migrateable in most cases.
